test/cmdlineargs.jl

```diff
@@ -145,12 +145,16 @@ end
 
 let exename = `$(Base.julia_cmd()) --startup-file=no --color=no`
     # tests for handling of ENV errors
-    let v = writereadpipeline(
-            "println(\"REPL: \", @which(less), @isdefined(InteractiveUtils))",
-            setenv(`$exename -i -E '@assert isempty(LOAD_PATH); push!(LOAD_PATH, "@stdlib"); @isdefined InteractiveUtils'`,
-                    "JULIA_LOAD_PATH" => "",
-                    "JULIA_DEPOT_PATH" => ";:",
-                    "HOME" => homedir()))
+    let
+        # Redirect stderr to devnull until the test is broken
+        redirect_stderr(devnull) do
```

Comment:
redirect the cmd's stderr using `pipeline`, rather than redirecting the parent process stdout?

Reply:
I added `stdout`/`stderr` keyword arguments to the helper function `writereadpipeline` so that we can more easily capture the streams and test them.